fontc/src/workload.rs

```diff
@@ -244,12 +254,110 @@ impl<'a> Workload<'a> {
                 }
             })
             .collect();
+        launchable.sort_by_key(|id| id.priority());
         trace!("Launchable: {launchable:?}");
 
         self.timing.add(timing.complete());
         launchable
     }
 
+    fn launch(
+        &mut self,
+        scope: &Scope,
+        send: &Sender<(AnyWorkId, Result<(), AnyWorkError>, JobTime)>,
+        abort: Arc<AtomicBool>,
+        fe_root: &FeContext,
+        be_root: &BeContext,
+        work_ids: &[AnyWorkId],
+    ) {
+        // Place anything that needs launching on the run queue
+        for id in work_ids.iter() {
+            log::trace!("Start {:?}", id);
+            let timing = create_timer(id.clone());
+            let job = self.jobs_pending.get_mut(id).unwrap();
+            job.running = true;
+            let work = job
+                .work
+                .take()
+                .expect("{id:?} ready to run but has no work?!");
+            if !job.run {
+                if let Err(e) = send.send((id.clone(), Ok(()), JobTime::nop(id.clone()))) {
+                    log::error!("Unable to write nop {id:?} to completion channel: {e}");
+                    //FIXME: if we can't send messages it means the receiver has dropped,
+                    //which means we should... return? abort?
+                }
+                continue;
+            }
+            let work_context = AnyContext::for_work(
+                fe_root,
+                be_root,
+                id,
+                job.read_access.clone(),
+                job.write_access.clone(),
+            );
+
+            let timing = timing.queued();
+            let element = (abort.clone(), timing, work, work_context);
+            {
+                let mut pending = self.pending_execution.lock().unwrap();
+                match id.priority() {
+                    Priority::High => pending.push_front(element),
+                    Priority::Low => pending.push_back(element),
+                }
+            }
+        }
+
+        // Launch 1 processor per thing that became runnable
```

Comment:
is it "1 _processor_ per thing" or rather "1 _thread_ per thing"?

Reply:
It's a job or task that runs on some thread; will rephrase